### DISCUSSION TITLE
How to use the find() to get selected sets of devices

### DISCUSSION THREAD

**Discussion by brianreinhold** [pyusb/pyusb#453] [Q&A] [answered]
I currently do the following to find all devices with a product id of a given value:
```
        # Find what COPPA devices are attached to the USB bus
        devs:list[usb.core.Device] = None
        try:
            devs = list(usb.core.find(True, idProduct = COPPA_DEVICE_ID[0]))
            print("Number of devices ", len(devs))
```
However, I would like to be able to discover additional devices that have another product id.  

Can one call find() multiple times and concatenate a list of Devices?

**Comment by jonasmalacofilho** [comment, marked as answer]
> Can one call find() multiple times and concatenate a list of Devices?

That way, by calling `find` multiple times and appending all results to a list (which you can do with or without concatenation, it depends on whether you really need to get concrete lists out of each "partial" call to `find(find_all=True, ...)`).

---

But can I suggest you simply call `find(find_all=True)` once (with no product ID filter), and filter the results manually?  This _might_ be slightly more efficient, depending on the platform and backend in use.
